Testing/SystemTests/lib/systemtests/systemtesting.py

```diff
@@ -1023,7 +1072,7 @@ def __shouldTest(self, suite):
                 return False
         return True
 
-    def executeTests(self, tests_done=None):
+    def executeTests(self, tests_done=None, status_dict=None):
```

Comment:
In/out parameters tend to be avoided in Python as, unlike C++ (where they still should be minimized):
  - it's hard to tell whether the intention of the author was that it should be modified or that this was an accident 
  - returning multiple values from a Python function is trivial.

I know that `tests_done` is done like this too but can we avoid adding further arguments used like this? Can a local `status_dict` be created in this function and returned to the caller. The caller can then update their main dictionary with these results?